src/services/submission_processor.py

```diff
@@ -43,46 +45,28 @@ async def upload_to_storage(lei: str, submission_id: str, content: bytes, extens
         raise HTTPException(status_code=HTTPStatus.INTERNAL_SERVER_ERROR, detail="Failed to upload file")
 
 
-async def validate_submission(lei: str, submission_id: str, content: bytes, background_tasks: BackgroundTasks):
+async def validate_submission(lei: str, submission: SubmissionDAO, content: bytes, background_tasks: BackgroundTasks):
     df = pd.read_csv(BytesIO(content), dtype=str, na_filter=False)
     validator_version = imeta.version("regtech-data-validator")
-
+    submission.state = SubmissionState.VALIDATION_IN_PROGRESS
+    submission.validation_ruleset_version = validator_version
     # Set VALIDATION_IN_PROGRESS
-    await update_submission(
-        SubmissionDAO(
-            submitter=submission_id,
-            state=SubmissionState.VALIDATION_IN_PROGRESS,
-            validation_ruleset_version=validator_version,
-        )
-    )
-    background_tasks.add_task(validate_and_update_submission, df, lei, submission_id, validator_version)
+    submission = await update_submission(submission)
+    background_tasks.add_task(validate_and_update_submission, df, lei, submission)
```

Comment:
I don't think this needs to be a task to just update the status anymore, once the file's uploaded to storage, we can say validation has started, then this method would just merge with the validate_and_update; don't think we need to have 2 bg tasks. (submission_uploaded will be valid once we have a separate microservice that listens on s3 events to actually do the validation much further down the line, but for now, I think we can just skip it)